flink-cdc-cli/src/main/java/org/apache/flink/cdc/cli/CliExecutor.java

```diff
@@ -54,22 +62,31 @@ public CliExecutor(
     }
 
     public PipelineExecution.ExecutionInfo run() throws Exception {
-        // Parse pipeline definition file
-        PipelineDefinitionParser pipelineDefinitionParser = new YamlPipelineDefinitionParser();
-        PipelineDef pipelineDef =
-                pipelineDefinitionParser.parse(pipelineDefPath, globalPipelineConfig);
-
-        // Create composer
-        PipelineComposer composer = getComposer(flinkConfig);
-
-        // Compose pipeline
-        PipelineExecution execution = composer.compose(pipelineDef);
-
-        // Execute the pipeline
-        return execution.execute();
+        // Create Submit Executor to deployment flink cdc job Or Run Flink CDC Job
+        boolean isDeploymentMode = ConfigurationUtils.isDeploymentMode(commandLine);
+        if (isDeploymentMode) {
+            ComposeExecutorFactory composeExecutorFactory = new ComposeExecutorFactory();
+            PipelineComposeExecutor composeExecutor =
+                    composeExecutorFactory.getFlinkComposeExecutor(commandLine);
```

Comment:
It's better to provided static method like `createFlinkComposeExecutor` instread of creating Factory.